### ISSUE TITLE
Under Docker, ping plugin not working for internet servers (local servers work)

### ISSUE THREAD

**Issue by darkguy2008** [influxdata/telegraf#6620]
# Relevant telegraf.conf:

```toml
[agent]
    interval = "10s"
    round_interval = true
    metric_batch_size = 1000
    metric_buffer_limit = 10000
    collection_jitter = "0s"
    flush_interval = "10s"
    flush_jitter = "0s"
    precision = ""
    debug = true
    quiet = false
    logfile = ""
    hostname = ""
    omit_hostname = false

[[outputs.influxdb]]
    urls = ["http://localhost:8086"]
    database = "telegraf"
    timeout = "30s"

[[inputs.ping]]
    urls = ["1.1.1.1"]
    count = 1
    timeout = 1.0
    interface = "enx00ce39c88877"
```

### System info:

Telegraf should be 1.7, 1.7.8, latest (per Dockerhub)
OS is Debian 10 (also tried on Pi 3's Debian Buster distro)
setcap with /bin/ping has been ran, no difference

### Steps to reproduce:

1. Use this docker-compose.yml file
2. Use this telegraf.conf file
3. Wait for it to run, this log happens:
```
influxdb_1  | [httpd] 172.18.0.1 - - [06/Nov/2019:05:52:10 +0000] "POST /write?db=telegraf HTTP/1.1" 204 0 "-" "Telegraf/1.12.4" 929c300c-0059-11ea-80aa-0242ac120002 28287
telegraf_1  | 2019-11-06T05:52:10Z D! [outputs.influxdb] Wrote batch of 1 metrics in 30.152622ms
telegraf_1  | 2019-11-06T05:52:10Z D! [outputs.influxdb] Buffer fullness: 0 / 10000 metrics
telegraf_1  | 2019-11-06T05:52:11Z W! [agent] [inputs.ping] did not complete within its interval
```
Special note to **W! [agent] [inputs.ping] did not complete within its interval**. So, packet loss is 100% for Telegraf, but ping works:
```
ping -c 1 -W 1.0 -I enx00ce39c88877 1.1.1.1
PING 1.1.1.1 (1.1.1.1) from 192.168.0.200 enx00ce39c88877: 56(84) bytes of data.
64 bytes from 1.1.1.1: icmp_seq=1 ttl=47 time=123 ms

--- 1.1.1.1 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 122.987/122.987/122.987/0.000 ms
```

### Expected behavior:

Should receive data in maximum_response_ms, minimum_response_ms and packets_received > 0

### Actual behavior:

All fields empty except name, time, host. packet_loss = 100

### Additional info:

docker-compose.yml
```
version: '3'
services:

  influxdb:
    image: influxdb:latest
    volumes:
      - ./data/influxdb:/var/lib/influxdb
    ports:
      - 8086:8086

  telegraf:
    image: telegraf:latest
    volumes:
      - ./telegraf.conf:/etc/telegraf/telegraf.conf:ro
    depends_on:
      - influxdb
    network_mode: host
```

**Comment by darkguy2008** [comment]
FOUND A FIX:

Okay, so screw all the parameters. Considering "ping" worked flawlessly, I used "arguments" instead.

What a mess, spent like 2 days on this...

Working telegraf.conf:
```
[[outputs.influxdb]]
    urls = ["http://localhost:8086"]
    database = "telegraf"
    timeout = "30s"

[[inputs.ping]]
    ipv6 = false
    method = "exec"
    urls = ["1.1.1.1"]
    arguments = ["-c", "1", "-W", "1.0", "-I", "enx00ce39c88877"]
```
I'm going to leave it open so somebody can fix this in the future and no need to use arguments anymore.

**Comment by danielnelson** [comment]
I think based on your config you actually have Telegraf 1.12.x.

Once 1.12.5 is released (next week most likely), as we currently have a bug in 1.12.4 with native ping, I highly recommend trying the `method = "native", it should behave less cryptically with respect to timeouts and be much less expensive to run.

I am curious what the exact problem was here, I think it is perhaps because we are passing `-w 10` by default to the ping executable.  Would you be able to check if temporarily adding this back to your arguments causes the issue to reappear?
